diff -ruN a/type-overrides/@wordpress/core-data/dynamic-entities.d.ts b/type-overrides/@wordpress/core-data/dynamic-entities.d.ts
--- a/type-overrides/@wordpress/core-data/dynamic-entities.d.ts
+++ b/type-overrides/@wordpress/core-data/dynamic-entities.d.ts
@@ -60,10 +60,10 @@
 	};
 	type DeleteRecordsHttpQuery = Record<string, any>;
 	export type SaveActions = {
-	    [Key in `save${keyof WPEntityTypes}`]: (data: Partial<WPEntityTypes[Key extends `save${infer E}` ? E : never]>, options?: ActionOptions) => Promise<void>;
+	    [Key in `save${keyof WPEntityTypes}`]: (data: Partial<WPEntityTypes[Key extends `save${infer E}` ? E : never]>, options?: ActionOptions) => (thunkArgs: any) => Promise<WPEntityTypes[Key extends `save${infer E}` ? E : never] | undefined>;
 	};
 	export type DeleteActions = {
-	    [Key in `delete${keyof WPEntityTypes}`]: (id: number | string, query?: DeleteRecordsHttpQuery, options?: ActionOptions) => Promise<void>;
+	    [Key in `delete${keyof WPEntityTypes}`]: (id: number | string, query?: DeleteRecordsHttpQuery, options?: ActionOptions) => (thunkArgs: any) => Promise<WPEntityTypes[Key extends `delete${infer E}` ? E : never] | false | undefined>;
 	};
 	export declare let dynamicActions: SaveActions & DeleteActions;
 	export declare let dynamicSelectors: SingularGetters & PluralGetters;
diff -ruN a/type-overrides/@wordpress/core-data/hooks/use-entity-prop.d.ts b/type-overrides/@wordpress/core-data/hooks/use-entity-prop.d.ts
--- a/type-overrides/@wordpress/core-data/hooks/use-entity-prop.d.ts
+++ b/type-overrides/@wordpress/core-data/hooks/use-entity-prop.d.ts
@@ -16,5 +16,5 @@
 	 * 							  information like `raw`, `rendered` and
 	 * 							  `protected` props.
 	 */
-	export default function useEntityProp(kind: string, name: string, prop: string, _id?: number | string): [any, Function, any];
+	export default function useEntityProp<PropType>(kind: string, name: string, prop: string, _id?: number | string): [PropType, (value: PropType) => void, PropType];
 }
diff -ruN a/type-overrides/@wordpress/core-data/index.d.ts b/type-overrides/@wordpress/core-data/index.d.ts
--- a/type-overrides/@wordpress/core-data/index.d.ts
+++ b/type-overrides/@wordpress/core-data/index.d.ts
@@ -80,156 +80,156 @@
 	    }) => Promise<void>;
 	    saveMenu: (data: Partial<import("@wordpress/core-data/build-types/entity-types").NavMenu<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").NavMenu<"edit"> | undefined>;
 	    saveComment: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Comment<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Comment<"edit"> | undefined>;
 	    saveGlobalStyles: (data: Partial<import("@wordpress/core-data/build-types/entity-types").GlobalStylesRevision<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").GlobalStylesRevision<"edit"> | undefined>;
 	    saveMedia: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Attachment<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Attachment<"edit"> | undefined>;
 	    saveMenuItem: (data: Partial<import("@wordpress/core-data/build-types/entity-types").NavMenuItem<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").NavMenuItem<"edit"> | undefined>;
 	    saveMenuLocation: (data: Partial<import("@wordpress/core-data/build-types/entity-types").MenuLocation<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").MenuLocation<"edit"> | undefined>;
 	    savePlugin: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Plugin<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Plugin<"edit"> | undefined>;
 	    savePostType: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Type<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Type<"edit"> | undefined>;
 	    saveRevision: (data: Partial<import("@wordpress/core-data/build-types/entity-types").PostRevision<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").PostRevision<"edit"> | undefined>;
 	    saveSidebar: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Sidebar<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Sidebar<"edit"> | undefined>;
 	    saveSite: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Settings<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Settings<"edit"> | undefined>;
 	    saveStatus: (data: Partial<import("@wordpress/core-data/build-types/entity-types").PostStatusObject<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").PostStatusObject<"edit"> | undefined>;
 	    saveTaxonomy: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Taxonomy<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Taxonomy<"edit"> | undefined>;
 	    saveTerm: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Term<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Term<"edit"> | undefined>;
 	    saveTheme: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Theme<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Theme<"edit"> | undefined>;
 	    saveUnstableBase: (data: Partial<import("@wordpress/core-data/build-types/entity-types").UnstableBase<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").UnstableBase<"edit"> | undefined>;
 	    saveUser: (data: Partial<import("@wordpress/core-data/build-types/entity-types").User<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").User<"edit"> | undefined>;
 	    saveWidget: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Widget<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Widget<"edit"> | undefined>;
 	    saveWidgetType: (data: Partial<import("@wordpress/core-data/build-types/entity-types").WidgetType<"edit">>, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").WidgetType<"edit"> | undefined>;
 	    deleteMenu: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").NavMenu<"edit"> | false | undefined>;
 	    deleteComment: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Comment<"edit"> | false | undefined>;
 	    deleteGlobalStyles: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").GlobalStylesRevision<"edit"> | false | undefined>;
 	    deleteMedia: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Attachment<"edit"> | false | undefined>;
 	    deleteMenuItem: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").NavMenuItem<"edit"> | false | undefined>;
 	    deleteMenuLocation: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").MenuLocation<"edit"> | false | undefined>;
 	    deletePlugin: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Plugin<"edit"> | false | undefined>;
 	    deletePostType: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Type<"edit"> | false | undefined>;
 	    deleteRevision: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").PostRevision<"edit"> | false | undefined>;
 	    deleteSidebar: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Sidebar<"edit"> | false | undefined>;
 	    deleteSite: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Settings<"edit"> | false | undefined>;
 	    deleteStatus: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").PostStatusObject<"edit"> | false | undefined>;
 	    deleteTaxonomy: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Taxonomy<"edit"> | false | undefined>;
 	    deleteTerm: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Term<"edit"> | false | undefined>;
 	    deleteTheme: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Theme<"edit"> | false | undefined>;
 	    deleteUnstableBase: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").UnstableBase<"edit"> | false | undefined>;
 	    deleteUser: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").User<"edit"> | false | undefined>;
 	    deleteWidget: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").Widget<"edit"> | false | undefined>;
 	    deleteWidgetType: (id: number | string, query?: {
 	        [x: string]: any;
 	    }, options?: {
 	        throwOnError?: boolean;
-	    }) => Promise<void>;
+	    }) => (thunkArgs: any) => Promise<import("@wordpress/core-data/build-types/entity-types").WidgetType<"edit"> | false | undefined>;
 	}, {
 	    getAuthors(state: selectors.State, query?: selectors.GetRecordsHttpQuery): import("@wordpress/core-data/build-types/entity-types").User[];
 	    getCurrentUser(state: selectors.State): import("@wordpress/core-data/build-types/entity-types").User<"view">;
@@ -317,12 +317,12 @@
 	    getPlugin: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Plugin<"edit"> | undefined;
 	    getPostType: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Type<"edit"> | undefined;
 	    getSidebar: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Sidebar<"edit"> | undefined;
-	    getSite: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Settings<"edit"> | undefined;
+	    getSite: (state: selectors.State, id?: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Settings<"edit"> | undefined;
 	    getStatus: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").PostStatusObject<"edit"> | undefined;
 	    getTaxonomy: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Taxonomy<"edit"> | undefined;
 	    getTerm: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Term<"edit"> | undefined;
 	    getTheme: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Theme<"edit"> | undefined;
-	    getUnstableBase: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").UnstableBase<"edit"> | undefined;
+	    getUnstableBase: (state: selectors.State, id?: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").UnstableBase<"edit"> | undefined;
 	    getUser: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").User<"edit"> | undefined;
 	    getWidget: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Widget<"edit"> | undefined;
 	    getWidgetType: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").WidgetType<"edit"> | undefined;
@@ -351,4 +351,10 @@
 	export * from "@wordpress/core-data/build-types/hooks";
 	export * from "@wordpress/core-data/build-types/private-apis";
 	import * as selectors from '@wordpress/core-data/build-types/selectors';
+}
+
+declare module '@wordpress/data/build-types/types' {
+	interface StoreRegistry {
+		'core': typeof import("@wordpress/core-data").store;
+	}
 }
diff -ruN a/type-overrides/@wordpress/core-data/index.d.ts.orig b/type-overrides/@wordpress/core-data/index.d.ts.orig
--- a/type-overrides/@wordpress/core-data/index.d.ts.orig
+++ b/type-overrides/@wordpress/core-data/index.d.ts.orig
@@ -0,0 +1,354 @@
+/// <reference path="./entity-provider.d.ts" />
+/// <reference path="./entity-types/index.d.ts" />
+/// <reference path="./fetch/index.d.ts" />
+/// <reference path="./hooks/index.d.ts" />
+/// <reference path="./private-apis.d.ts" />
+/// <reference path="./selectors.d.ts" />
+
+declare module '@wordpress/core-data' {
+	/**
+	 * Store definition for the code data namespace.
+	 *
+	 * @see https://github.com/WordPress/gutenberg/blob/HEAD/packages/data/README.md#createReduxStore
+	 */
+	export const store: import("@wordpress/data/build-types/types").StoreDescriptor<import("@wordpress/data/build-types/types").ReduxStoreConfig<any, {
+	    __unstableAcquireStoreLock: (store: any, path: any, { exclusive }: {
+	        exclusive: any;
+	    }) => () => Promise<any>;
+	    __unstableReleaseStoreLock: (lock: any) => () => void;
+	    receiveUserQuery(queryID: string, users: any[] | any): any;
+	    receiveCurrentUser(currentUser: any): any;
+	    addEntities(entities: any[]): any;
+	    receiveEntityRecords(kind: string, name: string, records: any[] | any, query: any | null, invalidateCache: (boolean | null) | undefined, edits: any | null, meta: any | null): any;
+	    receiveCurrentTheme(currentTheme: any): any;
+	    __experimentalReceiveCurrentGlobalStylesId(currentGlobalStylesId: string): any;
+	    __experimentalReceiveThemeBaseGlobalStyles(stylesheet: string, globalStyles: any): any;
+	    __experimentalReceiveThemeGlobalStyleVariations(stylesheet: string, variations: any[]): any;
+	    receiveThemeSupports(): any;
+	    receiveThemeGlobalStyleRevisions(currentId: number, revisions: any[]): any;
+	    receiveEmbedPreview(url: string, preview: any): any;
+	    receiveUploadPermissions(hasUploadPermissions: boolean): any;
+	    receiveUserPermission(key: string, isAllowed: boolean): any;
+	    receiveUserPermissions(permissions: {
+	        [x: string]: boolean;
+	    }): any;
+	    receiveAutosaves(postId: number, autosaves: any[] | any): any;
+	    receiveNavigationFallbackId(fallbackId: integer): any;
+	    receiveDefaultTemplateId(query: any, templateId: string): any;
+	    deleteEntityRecord: (kind: string, name: string, recordId: number | string, query: any | null, { __unstableFetch, throwOnError }?: {
+	        __unstableFetch?: Function | undefined;
+	        throwOnError?: boolean | undefined;
+	    }) => ({ dispatch, resolveSelect }: {
+	        dispatch: any;
+	        resolveSelect: any;
+	    }) => Promise<boolean | undefined>;
+	    editEntityRecord: (kind: string, name: string, recordId: number | string, edits: any, options?: {
+	        undoIgnore?: boolean | undefined;
+	    }) => any;
+	    undo: () => ({ select, dispatch }: {
+	        select: any;
+	        dispatch: any;
+	    }) => void;
+	    redo: () => ({ select, dispatch }: {
+	        select: any;
+	        dispatch: any;
+	    }) => void;
+	    __unstableCreateUndoLevel: () => any;
+	    saveEntityRecord: (kind: string, name: string, record: any, { isAutosave, __unstableFetch, throwOnError, }?: {
+	        isAutosave?: boolean | undefined;
+	        __unstableFetch?: Function | undefined;
+	        throwOnError?: boolean | undefined;
+	    }) => ({ select, resolveSelect, dispatch }: {
+	        select: any;
+	        resolveSelect: any;
+	        dispatch: any;
+	    }) => Promise<any>;
+	    __experimentalBatch: (requests: any[]) => (thunkArgs: any) => Promise<any>;
+	    saveEditedEntityRecord: (kind: string, name: string, recordId: any, options?: any | undefined) => ({ select, dispatch, resolveSelect }: {
+	        select: any;
+	        dispatch: any;
+	        resolveSelect: any;
+	    }) => Promise<any>;
+	    __experimentalSaveSpecifiedEntityEdits: (kind: string, name: string, recordId: number | string, itemsToSave: any[], options: any) => ({ select, dispatch, resolveSelect }: {
+	        select: any;
+	        dispatch: any;
+	        resolveSelect: any;
+	    }) => Promise<any>;
+	    receiveRevisions: (kind: string, name: string, recordKey: number | string, records: any[] | any, query: any | null, invalidateCache: (boolean | null) | undefined, meta: any | null) => ({ dispatch, resolveSelect }: {
+	        dispatch: any;
+	        resolveSelect: any;
+	    }) => Promise<void>;
+	    saveMenu: (data: Partial<import("@wordpress/core-data/build-types/entity-types").NavMenu<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveComment: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Comment<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveGlobalStyles: (data: Partial<import("@wordpress/core-data/build-types/entity-types").GlobalStylesRevision<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveMedia: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Attachment<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveMenuItem: (data: Partial<import("@wordpress/core-data/build-types/entity-types").NavMenuItem<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveMenuLocation: (data: Partial<import("@wordpress/core-data/build-types/entity-types").MenuLocation<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    savePlugin: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Plugin<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    savePostType: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Type<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveRevision: (data: Partial<import("@wordpress/core-data/build-types/entity-types").PostRevision<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveSidebar: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Sidebar<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveSite: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Settings<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveStatus: (data: Partial<import("@wordpress/core-data/build-types/entity-types").PostStatusObject<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveTaxonomy: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Taxonomy<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveTerm: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Term<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveTheme: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Theme<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveUnstableBase: (data: Partial<import("@wordpress/core-data/build-types/entity-types").UnstableBase<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveUser: (data: Partial<import("@wordpress/core-data/build-types/entity-types").User<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveWidget: (data: Partial<import("@wordpress/core-data/build-types/entity-types").Widget<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    saveWidgetType: (data: Partial<import("@wordpress/core-data/build-types/entity-types").WidgetType<"edit">>, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteMenu: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteComment: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteGlobalStyles: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteMedia: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteMenuItem: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteMenuLocation: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deletePlugin: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deletePostType: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteRevision: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteSidebar: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteSite: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteStatus: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteTaxonomy: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteTerm: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteTheme: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteUnstableBase: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteUser: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteWidget: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	    deleteWidgetType: (id: number | string, query?: {
+	        [x: string]: any;
+	    }, options?: {
+	        throwOnError?: boolean;
+	    }) => Promise<void>;
+	}, {
+	    getAuthors(state: selectors.State, query?: selectors.GetRecordsHttpQuery): import("@wordpress/core-data/build-types/entity-types").User[];
+	    getCurrentUser(state: selectors.State): import("@wordpress/core-data/build-types/entity-types").User<"view">;
+	    getEntitiesByKind(state: selectors.State, kind: string): Array<any>;
+	    getEntity(state: selectors.State, kind: string, name: string): any;
+	    getEntityConfig(state: selectors.State, kind: string, name: string): any;
+	    hasEntityRecord(state: selectors.State, kind: string, name: string, key?: string | number, query?: selectors.GetRecordsHttpQuery): boolean;
+	    __experimentalGetEntityRecordNoResolver<EntityRecord extends import("@wordpress/core-data/build-types/entity-types").EntityRecord<any>>(state: selectors.State, kind: string, name: string, key: string | number): EntityRecord | undefined;
+	    hasEntityRecords(state: selectors.State, kind: string, name: string, query?: selectors.GetRecordsHttpQuery): boolean;
+	    getEntityRecordEdits(state: selectors.State, kind: string, name: string, recordId: string | number): any;
+	    hasEditsForEntityRecord(state: selectors.State, kind: string, name: string, recordId: string | number): boolean;
+	    isAutosavingEntityRecord(state: selectors.State, kind: string, name: string, recordId: string | number): boolean;
+	    isSavingEntityRecord(state: selectors.State, kind: string, name: string, recordId: string | number): boolean;
+	    isDeletingEntityRecord(state: selectors.State, kind: string, name: string, recordId: string | number): boolean;
+	    getLastEntitySaveError(state: selectors.State, kind: string, name: string, recordId: string | number): any;
+	    getLastEntityDeleteError(state: selectors.State, kind: string, name: string, recordId: string | number): any;
+	    getUndoEdit(state: selectors.State): any;
+	    getRedoEdit(state: selectors.State): any;
+	    hasUndo(state: selectors.State): boolean;
+	    hasRedo(state: selectors.State): boolean;
+	    getCurrentTheme(state: selectors.State): any;
+	    __experimentalGetCurrentGlobalStylesId(state: selectors.State): string;
+	    getThemeSupports(state: selectors.State): any;
+	    getEmbedPreview(state: selectors.State, url: string): any;
+	    isPreviewEmbedFallback(state: selectors.State, url: string): boolean;
+	    canUser(state: selectors.State, action: string, resource: string | {
+	        kind: string;
+	        name: string;
+	        id?: string | number;
+	    }, id?: string | number): boolean | undefined;
+	    canUserEditEntityRecord(state: selectors.State, kind: string, name: string, recordId: string | number): boolean | undefined;
+	    getAutosaves(state: selectors.State, postType: string, postId: string | number): Array<any> | undefined;
+	    getAutosave<EntityRecord extends import("@wordpress/core-data/build-types/entity-types").EntityRecord<any>>(state: selectors.State, postType: string, postId: string | number, authorId: string | number): EntityRecord | undefined;
+	    getReferenceByDistinctEdits(state: any): any;
+	    __experimentalGetCurrentThemeBaseGlobalStyles(state: selectors.State): any;
+	    __experimentalGetCurrentThemeGlobalStylesVariations(state: selectors.State): string | null;
+	    getBlockPatterns(state: selectors.State): Array<any>;
+	    getBlockPatternCategories(state: selectors.State): Array<any>;
+	    getUserPatternCategories(state: selectors.State): Array<selectors.UserPatternCategory>;
+	    getCurrentThemeGlobalStylesRevisions(state: selectors.State): Array<object> | null;
+	    getDefaultTemplateId(state: selectors.State, query: {
+	        slug?: string;
+	        is_custom?: boolean;
+	        ignore_empty?: boolean;
+	    }): string;
+	    isRequestingEmbedPreview: {
+	        (state: selectors.State, url: string): boolean;
+	        isRegistrySelector?: boolean;
+	        registry?: any;
+	    };
+	    getUserQueryResults: ((state: selectors.State, queryID: string) => import("@wordpress/core-data/build-types/entity-types").User<"edit">[]) & import("rememo").EnhancedSelector;
+	    getEntitiesConfig: ((state: selectors.State, kind: string) => Array<any>) & import("rememo").EnhancedSelector;
+	    getEntityRecord: selectors.GetEntityRecord;
+	    getRawEntityRecord: (<EntityRecord extends import("@wordpress/core-data/build-types/entity-types").EntityRecord<any>>(state: selectors.State, kind: string, name: string, key: string | number) => EntityRecord | undefined) & import("rememo").EnhancedSelector;
+	    getEntityRecords: selectors.GetEntityRecords;
+	    getEntityRecordsTotalItems: (state: selectors.State, kind: string, name: string, query: selectors.GetRecordsHttpQuery) => number | null;
+	    getEntityRecordsTotalPages: (state: selectors.State, kind: string, name: string, query: selectors.GetRecordsHttpQuery) => number | null;
+	    __experimentalGetDirtyEntityRecords: ((state: selectors.State) => Array<{
+	        title: string;
+	        key: string | number;
+	        name: string;
+	        kind: string;
+	    }>) & import("rememo").EnhancedSelector;
+	    __experimentalGetEntitiesBeingSaved: ((state: selectors.State) => Array<{
+	        title: string;
+	        key: string | number;
+	        name: string;
+	        kind: string;
+	    }>) & import("rememo").EnhancedSelector;
+	    getEntityRecordNonTransientEdits: ((state: selectors.State, kind: string, name: string, recordId: string | number) => any) & import("rememo").EnhancedSelector;
+	    getEditedEntityRecord: (<EntityRecord extends import("@wordpress/core-data/build-types/entity-types").EntityRecord<any>>(state: selectors.State, kind: string, name: string, recordId: string | number) => import("@wordpress/core-data/build-types/entity-types").Updatable<EntityRecord> | false) & import("rememo").EnhancedSelector;
+	    hasFetchedAutosaves: {
+	        (state: selectors.State, postType: string, postId: string | number): boolean;
+	        isRegistrySelector?: boolean;
+	        registry?: any;
+	    };
+	    getRevisions: (state: selectors.State, kind: string, name: string, recordKey: string | number, query?: selectors.GetRecordsHttpQuery) => (Record<import("@wordpress/core-data/build-types/entity-types").Context, Record<number, import("@wordpress/core-data/build-types/entity-types").PostRevision>> | Record<import("@wordpress/core-data/build-types/entity-types").Context, Record<number, import("@wordpress/core-data/build-types/entity-types").GlobalStylesRevision>>)[] | null;
+	    getRevision: ((state: selectors.State, kind: string, name: string, recordKey: string | number, revisionKey: string | number, query?: selectors.GetRecordsHttpQuery) => (Record<import("@wordpress/core-data/build-types/entity-types").Context, Record<number, import("@wordpress/core-data/build-types/entity-types").PostRevision>> | Record<import("@wordpress/core-data/build-types/entity-types").Context, Record<number, import("@wordpress/core-data/build-types/entity-types").GlobalStylesRevision>>) | Record<PropertyKey, never> | undefined) & import("rememo").EnhancedSelector;
+	    getMenu: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").NavMenu<"edit"> | undefined;
+	    getComment: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Comment<"edit"> | undefined;
+	    getGlobalStyles: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").GlobalStylesRevision<"edit"> | undefined;
+	    getMedia: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Attachment<"edit"> | undefined;
+	    getMenuItem: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").NavMenuItem<"edit"> | undefined;
+	    getMenuLocation: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").MenuLocation<"edit"> | undefined;
+	    getPlugin: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Plugin<"edit"> | undefined;
+	    getPostType: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Type<"edit"> | undefined;
+	    getSidebar: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Sidebar<"edit"> | undefined;
+	    getSite: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Settings<"edit"> | undefined;
+	    getStatus: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").PostStatusObject<"edit"> | undefined;
+	    getTaxonomy: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Taxonomy<"edit"> | undefined;
+	    getTerm: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Term<"edit"> | undefined;
+	    getTheme: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Theme<"edit"> | undefined;
+	    getUnstableBase: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").UnstableBase<"edit"> | undefined;
+	    getUser: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").User<"edit"> | undefined;
+	    getWidget: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Widget<"edit"> | undefined;
+	    getWidgetType: (state: selectors.State, id: number | string, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").WidgetType<"edit"> | undefined;
+	    getMenus: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").NavMenu<"edit">[] | null;
+	    getComments: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Comment<"edit">[] | null;
+	    getMediaItems: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Attachment<"edit">[] | null;
+	    getMenuItems: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").NavMenuItem<"edit">[] | null;
+	    getMenuLocations: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").MenuLocation<"edit">[] | null;
+	    getPlugins: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Plugin<"edit">[] | null;
+	    getPostTypes: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Type<"edit">[] | null;
+	    getSidebars: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Sidebar<"edit">[] | null;
+	    getSites: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Settings<"edit">[] | null;
+	    getStatuses: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").PostStatusObject<"edit">[] | null;
+	    getTaxonomies: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Taxonomy<"edit">[] | null;
+	    getTerms: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Term<"edit">[] | null;
+	    getThemes: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Theme<"edit">[] | null;
+	    getUnstableBases: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").UnstableBase<"edit">[] | null;
+	    getUsers: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").User<"edit">[] | null;
+	    getWidgets: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").Widget<"edit">[] | null;
+	    getWidgetTypes: (state: selectors.State, query?: selectors.GetRecordsHttpQuery) => import("@wordpress/core-data/build-types/entity-types").WidgetType<"edit">[] | null;
+	}>>;
+	export { default as EntityProvider } from "@wordpress/core-data/build-types/entity-provider";
+	export * from "@wordpress/core-data/build-types/entity-provider";
+	export * from "@wordpress/core-data/build-types/entity-types";
+	export * from "@wordpress/core-data/build-types/fetch";
+	export * from "@wordpress/core-data/build-types/hooks";
+	export * from "@wordpress/core-data/build-types/private-apis";
+	import * as selectors from '@wordpress/core-data/build-types/selectors';
+}
