diff --git a/build/admin/create-new-post.js b/build/admin/create-new-post.js
index 262087b71c40cabfd60254da376414325c8b23db..e9f5cb382675f318f788c58141c3cd1b75e25c37 100644
--- a/build/admin/create-new-post.js
+++ b/build/admin/create-new-post.js
@@ -22,7 +22,11 @@ async function createNewPost(options = {}) {
     if (excerpt) {
         query.set('excerpt', excerpt);
     }
+	const waitForDocumentReady =
+		this.page.waitForLoadState( 'domcontentloaded' );
     await this.visitAdminPage('post-new.php', query.toString());
+	await waitForDocumentReady;
+
     await this.editor.setPreferences('core/edit-post', {
         welcomeGuide: options.showWelcomeGuide ?? false,
         fullscreenMode: options.fullscreenMode ?? false,
diff --git a/build/admin/visit-site-editor.js b/build/admin/visit-site-editor.js
index ca74a89371c245c317d2a827bf07e41127209462..84c7541a80c050801c62251294bd36c03e512ac5 100644
--- a/build/admin/visit-site-editor.js
+++ b/build/admin/visit-site-editor.js
@@ -22,8 +22,11 @@ async function visitSiteEditor(options = {}) {
     if (canvas) {
         query.set('canvas', canvas);
     }
+	const waitForDocumentReady =
+		this.page.waitForLoadState( 'domcontentloaded' );
     await this.visitAdminPage('site-editor.php', query.toString());
     if (!options.showWelcomeGuide) {
+		await waitForDocumentReady;
         await this.editor.setPreferences('core/edit-site', {
             welcomeGuide: false,
             welcomeGuideStyles: false,
