diff -ruN a/types/@wordpress/editor/index.d.ts b/types/@wordpress/editor/index.d.ts
--- a/types/@wordpress/editor/index.d.ts
+++ b/types/@wordpress/editor/index.d.ts
@@ -11,3 +11,9 @@
 	export * from "@wordpress/editor/build-types/dataviews/api";
 	export { storeConfig, store } from "@wordpress/editor/build-types/store";
 }
+
+declare module '@wordpress/data' {
+	interface StoreRegistry {
+		'core/editor': typeof import("@wordpress/editor/build-types/store").store;
+	}
+}
