diff -ruN a/types/@wordpress/notices/index.d.ts b/types/@wordpress/notices/index.d.ts
--- a/types/@wordpress/notices/index.d.ts
+++ b/types/@wordpress/notices/index.d.ts
@@ -3,3 +3,11 @@
 declare module '@wordpress/notices' {
 	export { store } from "@wordpress/notices/build-types/store";
 }
+
+declare module '@wordpress/data/build-types/types' {
+	import { store as noticesStore } from '@wordpress/notices';
+
+	interface StoreRegistry {
+		'core/notices': typeof noticesStore;
+	}
+}
